src/Stack/Docker.hs

```diff
@@ -340,6 +342,13 @@ runContainerAndExit = do
     mountArg mountSuffix (Mount host container) =
       ["-v",host ++ ":" ++ container ++ mountSuffix]
     sshRelDir = relDirDotSsh
+    toLinuxStylePath s | osIsWindows =
+      T.pack s
+      & T.replace ":\\" "/"
```

Comment:
This doesn't look right, it would convert `c:\\foo` into `c/foo`, for instance

Reply:
I will double check and then fix or try to convince you that this is the right thing to do.

Reply:
Together with `& addStartingSlashIfMissing` (line 352) `C:\\foo` becomes `/c/foo` which is a legal and working thing. Maybe I could share some output from how this is working on my Windows machine to convince you that the folder sharing feature is working as expected?

As legal and working does not imply right I'm ready to discuss alternatives. I could mount/share the folders in `/mnt/` (for our example `/mnt/c/foo`). What do you think about it?

Reply:
I honestly can't figure out how all of that is supposed to work out correctly. As it stands now, this seems like it's trying to slyly address a bunch of edge cases, and I'm not convinced it's handling them correctly. It would be better to start off with what the expected behavior is for these edge cases, such as `"c:\foo\bar"` or relative paths like `"foo\bar"`. It seems to me like these two cases are now in conflict.

Reply:
You are right - it may be only working thanks to the fact that all paths happened to be absolute in my scenario. I will add explicit handling for absolute/relative paths and try to remove this hacky/sly smell around this change.